src/backends/google-sheets-backend.ts

```diff
@@ -12,141 +13,89 @@ interface BookSpreadsheetRow {
     isbn: string,
     booktitle: string,
     numcopies: number,
-    borrowers: string
+    borrowers: string,
 }
 
 export class GoogleSheetsBackend implements IBackend {
-    private _dbWorksheetIndex = 1;
-
-    constructor(private _doc: GoogleSpreadsheet) {}
+    constructor(
+        private readonly doc: GoogleSpreadsheet,
+        private readonly addRow: (worksheet_id: number, new_row: any) => Promise<SpreadsheetRow>,
+        private readonly getRows: (sheetIndex: number, opts: GetRows) => Promise<SpreadsheetRow[]>,
+        private readonly dbWorksheetIndex = 1,
+    ) {
+    }
 
-    static create(): Promise<GoogleSheetsBackend> {
+    static async create(): Promise<GoogleSheetsBackend> {
```

Comment:
Should you always have `async` here if you're returning a `Promise`?

Reply:
nope, any function can return a promise, but an `async` function *must* return a promise, and can use the `await` keyword internally.